solana/programs/example-native-token-transfers/src/instructions/admin.rs

```diff
@@ -15,13 +15,19 @@ use crate::{
 
 // * Transfer ownership
 
-/// Transferring the ownership is a 2-step process. The first step is to set the
+/// For safety reasons, transferring ownership is a 2-step process. The first step is to set the
 /// new owner, and the second step is for the new owner to claim the ownership.
 /// This is to prevent a situation where the ownership is transferred to an
 /// address that is not able to claim the ownership (by mistake).
 ///
 /// The transfer can be cancelled by the existing owner invoking the [`claim_ownership`]
 /// instruction.
+///
+/// Alternatively, the ownership can be transferred in a single step by calling the
+/// [`transfer_ownership_one_step`] instruction. This can be dangerous because if the new owner
+/// cannot actually sign transactions (due to setting the wrong address), the program will be
```

Comment:
Very very minor point on this but what if we add a constraint such that `new_owner` cannot be `upgrade_lock`?

* In the normal two-step case, this is invalid but it can be (re)claimed by the current `owner`.

* In the one-step case, this bricks the owner field though not in a particularly special way as any invalid address will permanently lock the owner field, as you noted here.

That said, might as well eliminate one tiny footgun since it's wrong in both cases?

Reply:
I agree it makes no sense to directly set the owner to `upgrade_lock`, but wouldn't it make the code a little confusing to read? In that I think that would be a fairly arbitrary check, or at least I don't see the risk of anyone accidentally supplying that particular wrong address vs any other one

Reply:
I think if I was reading for the first time it's helpful to know if something is invalid especially if it has something like `Error::UpgradeLockCantBeOwner`

Clearly a subjective matter though :) So let's not do it